modules/realsense_d435.py

```diff
@@ -0,0 +1,192 @@
+import pyrealsense2 as rs
+import traceback
+import sys
+import time
+import numpy as np
+
+class rs_d435:
+    """
+    Declare all the constants, tunable variables are public
+    """
+    def __init__(self):
+        # public
+        self.depth_width = 640
+        self.depth_height = 480
+
+        self.rgb_width = 640
+        self.rgb_height = 480
+
+        self.framerate = 30
+
+        self.min_range = 0.1
+        self.max_range = 10
+
+        # private
+        self._pipe = None
+
+        self._intrin = None
+        self._scale = 1
+
+        self._FOV = (0, 0)
+
+        self._x_deproject_matrix = None
+        self._y_deproject_matrix = None
+
+    """
+    with enter method opens D435 connection
+    """
+    def __enter__(self):
+        self.open_connection()
+
+    """
+    with exit method closes the D435 connection
+    """
+    def __exit__(self, exception_type, exception_value, traceback):
+        if traceback:
+            print(traceback.tb_frame)
+
+        self.close_connection()
+    
+    """
+    Open Connection to D435 camera
+    """
+    def open_connection(self):
+        cfg = rs.config()
+        cfg.enable_stream(rs.stream.depth, 
+                                self.depth_width, self.depth_height,
+                                rs.format.z16, 
+                                self.framerate )
+        cfg.enable_stream(rs.stream.color, 
+                                self.rgb_width, self.rgb_height, 
+                                rs.format.bgr8, 
+                                self.framerate)
+
+        self._pipe = rs.pipeline()
+        self._pipe.start( cfg  )
+
+        self._initialise_deprojection_matrix()
+
+        print('rs_d435:D435 Connection Open')
+    
+    """
+    Close connection to D435 camera
+    """
+    def close_connection(self):
+        if self._pipe is None:
+            return
+
+        self._pipe.stop()
+        self._pipe = None
+
+        print('rs_t265:D435 Connection Closed')
```

Comment:
Shouldnt this be rs_d435:D435 Connection closed?

Reply:
Fixed in 8ffb4b6